### DISCUSSION TITLE
Using spherical harmonics p=1 breaks equivariance

### DISCUSSION THREAD

**Discussion by dangthatsright** [e3nn/e3nn#360] [Q&A] [answered]
Sorry for another question but I am puzzled by this behavior. I am using `from e3nn.nn.models.v2103.gate_points_networks import NetworkForAGraphWithAttributes` and I have a simple network with various `0e` node inputs, edge attr, node attr, and node outputs. I am basically creating an SE(3) network.  The edge embedding step in `NetworkForAGraphWithAttributes` embeds the edges to spherical harmonics of of order lmax. I figure I can add in `p=1` as an arg on line https://github.com/e3nn/e3nn/blob/main/e3nn/nn/models/v2103/gate_points_networks.py#L45 and line 90. 

However running this network through assert_equivariant produces very different outputs when the parity is flipped, whereas I would expect it to be invariant to parity. Can someone explain this behavior?

**Comment by mariogeiger** [comment, marked as answer]
Here is my guess of the mistake you are making. Tell me if I'm wrong.

The irreps of relative positions is `"1o"`.
The irreps of `spherical_harmonics("1o") = "0e+1o+2e+3o+..."`

If you impose that the output of your spherical harmonics is `"0e+1e+2e+3e+..."` but you still manipulate your relative position as vectors (`"1o"`) you break equivariance to parity.

You need to tell `assert_equivariant` that your positions are `"1e"` in order to not test parity. Or there might be a flag to test or not the parity if I'm not wrong.

**Reply by dangthatsright** [reply]
Thanks @mariogeiger that explains a lot. I always assumed that relative positions is `1e`. Can you give me an example of what something that is `1e`? Is it some sort of like half space of R^3 thing?

**Reply by mariogeiger** [reply]
For instance [the magnetic field is a pseudovector](https://physics.stackexchange.com/questions/352905/why-is-mathbfb-a-pseudovector).

**Reply by dangthatsright** [reply]
thanks! i'll read more about it
